lib/js_routes/instance.rb

```diff
@@ -24,9 +24,15 @@ def initialize(**options)
     sig {returns(String)}
     def generate
       # Ensure routes are loaded. If they're not, load them.
+
       application = T.unsafe(self.application)
-      if named_routes.empty? && application.respond_to?(:reload_routes!, true)
-        application.reload_routes!
+      if named_routes.empty?
+        rails8_or_newer = Gem::Version.new(Rails.version) >= Gem::Version.new('8')
+        if rails8_or_newer
```

Comment:
String comparison is enough:

```suggestion
        if Rails.version >= "8.0.0"
```